android/src/com/mapswithme/maps/editor/EditorFragment.java

```diff
@@ -330,30 +325,35 @@ private void refreshOpeningTime()
     }
   }
 
-  private void initLocalizedNameView(final View view)
+  private void initNamesView(final View view)
   {
-    mLocalizedNames = (RecyclerView) view.findViewById(R.id.recycler);
-    mLocalizedNames.setNestedScrollingEnabled(false);
-    mLocalizedNames.setLayoutManager(new LinearLayoutManager(getActivity()));
-    mLocalizedNamesAdapter = new MultilanguageAdapter(mParent);
-    mLocalizedNames.setAdapter(mLocalizedNamesAdapter);
-    mLocalizedNamesAdapter.registerAdapterDataObserver(mLocalizedNamesObserver);
-    refreshLocalizedNames();
+    mNamesCaption = (TextView) view.findViewById(R.id.show_additional_names);
+    mNamesCaption.setOnClickListener(this);
+    mAddLanguage = (TextView) view.findViewById(R.id.add_langs);
+    mAddLanguage.setOnClickListener(this);
+    mMoreLanguages = (TextView) view.findViewById(R.id.more_names);
+    mMoreLanguages.setOnClickListener(this);
+    mNamesView = (RecyclerView) view.findViewById(R.id.recycler);
+    mNamesView.setNestedScrollingEnabled(false);
+    mNamesView.setLayoutManager(new LinearLayoutManager(getActivity()));
+    mNamesAdapter = new MultilanguageAdapter(mParent);
+    mNamesView.setAdapter(mNamesAdapter);
+    mNamesAdapter.registerAdapterDataObserver(mNamesObserver);
 
     final Bundle args = getArguments();
-    if (args == null || !args.containsKey(LAST_LOCALIZED_NAME_INDEX))
+    if (args == null || !args.containsKey(LAST_INDEX_OF_NAMES_ARRAY))
```

Comment:
Why you don't show additional names when args does not have LAST_INDEX_OF_NAMES_ARRAY? Looks like LAST_INDEX_OF_NAMES_ARRAY is used to focus, right?